lib/home_page.dart

```diff
@@ -8,15 +24,288 @@ class HomePage extends StatefulWidget {
+                              'assets/chat_shrug.png',
+                              scale: 8,
+                            ),
+                            const SizedBox(height: 16),
+                            Text(
+                              S.of(context).pageHomePageEmpty,
+                              style: Theme.of(context).textTheme.headline6,
+                            ),
+                            const SizedBox(height: 3),
+                          ],
+                        ),
+                ),
+              )
+            else
+              const Center(
+                child: CircularProgressIndicator(),
+              ),
+            const SizedBox(height: 35),
+            Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                Hero(
+                  tag: 'SearchButton',
+                  transitionOnUserGestures: true,
+                  child: Button(
+                    S.of(context).pageHomeSearchButton,
+                    onPressed: () => Navigator.of(context).push(
+                      MaterialPageRoute<void>(builder: (BuildContext context) => const SearchRidePage()),
+                    ),
+                  ),
+                ),
+                const SizedBox(height: 15),
+                Hero(
+                  tag: 'createButton',
+                  transitionOnUserGestures: true,
+                  child: Button(
+                    S.of(context).pageHomeCreateButton,
+                    onPressed: () => Navigator.of(context).push(
+                      MaterialPageRoute<void>(builder: (BuildContext context) => const CreateDrivePage()),
+                    ),
+                  ),
+                ),
+                const SizedBox(
+                  height: 35,
+                )
+              ],
+            ),
+          ],
+        ),
       ),
     );
   }
+
+  Future<void> _loadNewRideEvent(int rideEventId) async {
+    final Map<String, dynamic> data = await SupabaseManager.supabaseClient.from('ride_events').select('''
+      *,
+      ride: ride_id(*,
+        rider: rider_id(*),
+        drive: drive_id(*,
+          driver: driver_id(*)
+          )
+      )
+      ''').eq('id', rideEventId).single();
+    final RideEvent rideEvent = RideEvent.fromJson(data);
+    if (rideEvent.isForCurrentUser()) {
+      setState(() {
+        _items.insert(0, rideEvent);
+      });
+    }
+  }
+
+  Future<void> _loadNewMessage(int messageId) async {
+    final Map<String, dynamic> data = await SupabaseManager.supabaseClient.from('messages').select('''
+      *,
+      sender: sender_id(*)
+      )
+    ''').eq('id', messageId).single();
+    final Message message = Message.fromJson(data);
+    if (message.read == false) {
+      setState(() {
+        _items.insert(0, message);
+      });
+    }
+  }
+
+  Card _buildMessageWidget(Message message, BuildContext context) {
+    return Card(
```

Comment:
there should be a way to dismiss these cards by swiping. That probably requires a new database column (because a message should not be marked as read just because you swiped). But we can do that in a follow-up PR.

Reply:
I think thats a great idea and I think we should do it in a different PR too.